features/lesson/src/main/kotlin/com/teacherworkout/features/lesson/LessonStartScreen.kt

```diff
@@ -0,0 +1,242 @@
+package com.teacherworkout.features.lesson
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Schedule
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.teacherworkout.commons.ui.model.Lesson
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import com.teacherworkout.features.account.composables.UpIcon
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.collect
+
+@Composable
+fun LessonStartScreen(
+    state: LessonContract.State,
+    onSendEvent: (LessonContract.Event) -> Unit,
+    effects: Flow<LessonContract.Effect>,
+    onNavigationRequest: (LessonContract.Effect.Navigation) -> Unit
+) {
+    val space16dp = dimensionResource(id = R.dimen.space_16dp)
+    val space24dp = dimensionResource(id = R.dimen.space_24dp)
+
+     LaunchedEffect(Unit) {
+        effects.onEach { effect ->
+            when (effect) {
+                is LessonContract.Effect.Navigation.NavigateUp -> {
+                    onNavigationRequest(effect)
+                }
+            }
+        }.collect()
+    }
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                backgroundColor = colorResource(R.color.landing_background),
+                elevation = 0.dp
+            ) {
+                //TODO: in terms of design, maybe this isn't the best way to handle the up navigation
```

Comment:
Not sure what you are referring to here exactly? We can discuss with the design team if there are any other ideas?

Reply:
I wrote that because in the Figma design to close the lesson start screen you would press on an 'x' in the upper right corner. As far as I know, that's an iOS thing. But I wasn't sure about what would be the equivalent in Android.

Reply:
I agree with that. The design it's created for both apps: android and iOS but we can adapt it to be more androidish